R/derive_param_clinbenefit.R

```diff
@@ -0,0 +1,256 @@
+#' @param source_datasets  A named list of data sets is expected.
+#'
+#'  The list must contain the names provided by the `dataset_name` field of the
+#'  `admiral::date_source()` objects specified for `source_pd` and `source_resp`.
+#'
+#' @param by_vars A named list returned by `vars()` used together with `subject_keys`
+#' to identify groupings within which the earliest date of clinical benefit rate
+#' status will be chosen.
+#'
+#' @param reference_date Name of variable representing the index date for
+#' `ref_start_window`. A variable providing a date. An unquoted symbol is expected.
+#'
+#' @param ref_start_window Integer representing number of days from `reference_date`
+#' that must elapse before an evaluable non-PD assessment counts toward determining
+#' clinical benefit.
+#'
+#' @param set_values_to A named list returned by `vars()` containing new variables
+#' and their static value to be populated for the clinical benefit rate parameter
+#' records, e.g. `vars(PARAMCD = "CBR", PARAM = "Clinical Benefit Rate")`.
+#'
+#' @param subject_keys A named list returned by `vars()` containing variables
+#' used to uniquely identify subjects.
+#'
+#' @export
+#'
+#' @author Andrew Smith
+#' @examples
+#' library(lubridate)
+#' library(dplyr)
+#' adsl <- tibble::tribble(
+#'   ~USUBJID, ~TRTSDT, ~EOSDT,
+#'   "01", ymd("2020-01-14"), ymd("2020-05-06"),
+#'   "02", ymd("2021-02-16"), ymd("2021-08-03"),
+#'   "03", ymd("2021-03-09"), ymd("2021-04-24"),
+#'   "04", ymd("2021-04-21"), ymd("2021-09-15")
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' adrs <- tibble::tribble(
+#'   ~USUBJID, ~PARAMCD, ~AVALC, ~ADT,
+#'   "01", "RSP", "Y", ymd("2021-03-14"),
+#'   "02", "RSP", "N", ymd("2021-05-07"),
+#'   "03", "RSP", "N", NA,
+#'   "04", "RSP", "N", NA,
+#'   "01", "PD", "N", NA,
+#'   "02", "PD", "Y", ymd("2021-05-07"),
+#'   "03", "PD", "N", NA,
+#'   "04", "PD", "N", NA,
+#'   "01", "OVR", "SD", ymd("2020-03-14"),
+#'   "01", "OVR", "PR", ymd("2021-04-13"),
+#'   "02", "OVR", "PR", ymd("2021-04-08"),
+#'   "02", "OVR", "PD", ymd("2021-05-07"),
+#'   "02", "OVR", "CR", ymd("2021-06-20"),
+#'   "03", "OVR", "SD", ymd("2021-03-30"),
+#'   "04", "OVR", "NE", ymd("2021-05-21"),
+#'   "04", "OVR", "NA", ymd("2021-06-30"),
+#'   "04", "OVR", "NE", ymd("2021-07-24"),
+#'   "04", "OVR", "ND", ymd("2021-09-04"),
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' pd <- admiral::date_source(
+#'   dataset_name = "adrs",
+#'   date = ADT,
+#'   filter = PARAMCD == "PD" & AVALC == "Y"
+#' )
+#'
+#' resp <- admiral::date_source(
+#'   dataset_name = "adrs",
+#'   date = ADT,
+#'   filter = PARAMCD == "RSP" & AVALC == "Y"
+#' )
+#'
+#' derive_param_clinbenefit(
+#'   dataset = adrs,
+#'   dataset_adsl = adsl,
+#'   filter_source = PARAMCD == "OVR",
+#'   source_resp = resp,
+#'   source_pd = pd,
+#'   source_datasets = list(adrs = adrs),
+#'   reference_date = TRTSDT,
+#'   ref_start_window = 28,
+#'   set_values_to = vars(
+#'     PARAMCD = "CBR"
+#'   )
+#' )
```

Comment:
The new parameter is not displayed in the rendered output. Maybe just display the new observations.

Reply:
updated to filter